test/test.js

```diff
@@ -1,131 +1,165 @@
-#!/usr/bin/env node
-/*
- * MozIoT Gateway App.
- *
- * Back end main script.
- *
- * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/.
- */
-
 'use strict';
 
-var Gateway = require('./gateway');
-var GetOpt = require('node-getopt');
-var utils = require('../utils');
-
-// Command line arguments
-var getopt = new GetOpt([
-  ['h', 'help',     'Display help' ],
-  ['v', 'verbose',  'Show verbose output' ],
-  ['l', 'list',     'List all things'],
-  ['',  'toggle',   'Toggle all onOff devices'],
-  ['', 'thing=NAME',     'Filter things that match NAME'],
-  ['', 'set=PROPERTY=value', 'Set all properties to a value'],
-]);
-
-getopt.bindHelp(
-  'Usage: node toggle.js SERVER\n' +
-  '\n' +
-  '[[OPTIONS]]'
-);
-var opt = getopt.parseSystem();
-const VERBOSE = opt.options.verbose;
-
-if (VERBOSE) {
-  console.log(opt);
-}
+/* Tell jshint about mocha globals, and  */
+/* globals describe, it */
 
-if (opt.argv.length != 1) {
-  console.log('Must specify exactly one server\n');
-  getopt.showHelp();
-  process.exit(1);
-}
+process.env.NODE_ENV = 'test';
 
-var server_url = opt.argv[0];
+let assert = require('assert');
+let chai = require('chai');
+let chaiHttp = require('chai-http');
+let server = require('../app');
 
-// Get some decent error messages for unhandled rejections. This is
-// often just errors in the code.
-process.on('unhandledRejection', (reason) => {
-  console.log('Unhandled Rejection');
-  console.error(reason);
-});
+let config = require('../config');
```

Comment:
It looks like if we use the config module: https://github.com/lorenwest/node-config then it allows for default configuration settings and overrides for production/development/test based on the NODE_ENV environment variable.

So I'm going to suggest that we switch to that.